cmd/telemetry-metrics-filter/main.go

```diff
@@ -0,0 +1,204 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/signal"
+	"strings"
+	"sync"
+	"syscall"
+
+	"github.com/namsral/flag"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+
+	// Needed for profiling
+	_ "net/http/pprof"
+)
+
+type BrokerHealthStatus string
+
+const (
+	BrokerHealthUnknown BrokerHealthStatus = "Unknown"
+	BrokerHealthClosed  BrokerHealthStatus = "Closed"
+	BrokerHealthError   BrokerHealthStatus = "Error"
+	BrokerHealthOk      BrokerHealthStatus = "Ok"
+)
+
+type BrokerHealth struct {
+	Status        BrokerHealthStatus `json:"Status"`
+	LastError     *string            `json:"LastError,omitempty"`
+	LastErrorCode *string            `json:"LastErrorCode,omitempty"`
+}
+
+var (
+	logger      *zap.Logger
+	atomicLevel zap.AtomicLevel
+)
+
+func setupLogging() {
+	logLevel := os.Getenv("LOG_LEVEL")
+	logLevel = strings.ToUpper(logLevel)
+
+	atomicLevel = zap.NewAtomicLevel()
+
+	encoderCfg := zap.NewProductionEncoderConfig()
+	logger = zap.New(zapcore.NewCore(
+		zapcore.NewJSONEncoder(encoderCfg),
+		zapcore.Lock(os.Stdout),
+		atomicLevel,
+	))
+
+	switch logLevel {
+	case "DEBUG":
+		atomicLevel.SetLevel(zap.DebugLevel)
+	case "INFO":
+		atomicLevel.SetLevel(zap.InfoLevel)
+	case "WARN":
+		atomicLevel.SetLevel(zap.WarnLevel)
+	case "ERROR":
+		atomicLevel.SetLevel(zap.ErrorLevel)
+	case "FATAL":
+		atomicLevel.SetLevel(zap.FatalLevel)
+	case "PANIC":
+		atomicLevel.SetLevel(zap.PanicLevel)
+	default:
+		atomicLevel.SetLevel(zap.InfoLevel)
+	}
+}
+
+func main() {
+	// Parse CLI flag configuration
+	brokerConfigFile := flag.String("broker_config_file", "./resources/telemetry-filter-broker-config-go.json", "Broker configuration file")
+	workerCount := flag.Int("worker_count", 10, "Number of event workers")
+	httpListenString := flag.String("http_listen", "0.0.0.0:9088", "HTTP Server listen string")
+	unmarshalEventStrategyString := flag.String("unmarshal_event_strategy", "hmcollector", "How should json payloads be unmarshaled")
```

Comment:
Right now the chart is overriding this to `go-json`. Which seems to be the fastest of the strategies. the default should change to it.